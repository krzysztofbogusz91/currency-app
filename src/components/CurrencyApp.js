import React, { Component } from 'react';
import data from '../api/nbp_tab';
import Table from './Table';
import { Header } from '../containers/header';

// <ul>
//     <li>Current buy and sell tab with currencies</li>
//     <li>Custom currency Y search with X last buy and sell options list</li>
//     <li>Current Time</li>
// </ul>

class CurrencyApp extends Component {
    constructor() {
       super()
       this.state = {
           list: []
       }
    }

    componentDidMount() {
        data()
            .then((result)=>{
                const list = result;
                this.setState ({ list }) 
            })
            .catch( err => {
                this.setState ({ list: [] })
            })
        // where to put reject in async/await
        // (async () => {
        //     const list = await data();
        //     this.setState ({ list })
        //     console.log(this.state.list)
        // })()
    }
    
    render() {
        return (
            <div>
                <Header />
                <div>
                    <Table list={this.state.list}/>
                </div>
                <footer>
                    <div className="text-right">timmer: 10:45:22</div>
                </footer>  
            </div>
        );
    }
}

export default CurrencyApp;
