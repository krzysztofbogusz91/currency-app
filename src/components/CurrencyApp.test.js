import React from 'react';
import CurrencyApp from './CurrencyApp';
import {shallow} from 'enzyme';
import { config } from '../enzyme_set_up';
import mockData from 'axios';

//why mount brakes it all? 
describe('CurrencyApp',()=>{
  let currency = shallow(<CurrencyApp />);
  describe('renders correctly and contain children components', () => {
      it('renders correctly', ()=>{
        expect(currency).toMatchSnapshot();
      })
      it('should contain `Table` component', () => {
        expect(currency.find('Table').exists()).toBe(true);
      });
  });
  
  describe('component `state` contain list and data after mock api call', () => {

      it('should have `state` list property', () => {
        expect(currency.state('list')).toBeTruthy();
      });
     
      it('`state` list should have length of two after calling api and containd code USD', () => {
        expect(currency.state('list').length).toEqual(2);
        expect(currency.state('list')[0]['code']).toEqual('USD')
      });

      it('`state` should have empty list on err', () => {
        //change return form mock call only once
        mockData.get.mockImplementationOnce(()=> Promise.reject({
          data: "error"
        })) 
        //renders again with new state
        let currency = shallow(<CurrencyApp />);
        expect(currency.state('list')).toEqual([])
        
      });
  });
})
