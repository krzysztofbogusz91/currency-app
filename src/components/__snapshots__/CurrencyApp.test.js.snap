// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyApp renders correctly and contain children components renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyApp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown />,
        <div>
          <Table
                    list={Array []}
          />
          <Form
                    getDataFromForm={[Function]}
                    list={Array []}
                    options={Array []}
          />
</div>,
        <footer>
          <div
                    className="text-right"
          >
                    timmer: 10:45:22
          </div>
</footer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Table
              list={Array []}
/>,
            <Form
              getDataFromForm={[Function]}
              list={Array []}
              options={Array []}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "list": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "getDataFromForm": [Function],
              "list": Array [],
              "options": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="text-right"
>
            timmer: 10:45:22
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "timmer: 10:45:22",
            "className": "text-right",
          },
          "ref": null,
          "rendered": "timmer: 10:45:22",
          "type": "div",
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <div>
            <Table
                        list={Array []}
            />
            <Form
                        getDataFromForm={[Function]}
                        list={Array []}
                        options={Array []}
            />
</div>,
          <footer>
            <div
                        className="text-right"
            >
                        timmer: 10:45:22
            </div>
</footer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Table
                list={Array []}
/>,
              <Form
                getDataFromForm={[Function]}
                list={Array []}
                options={Array []}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "list": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "getDataFromForm": [Function],
                "list": Array [],
                "options": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="text-right"
>
              timmer: 10:45:22
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "timmer: 10:45:22",
              "className": "text-right",
            },
            "ref": null,
            "rendered": "timmer: 10:45:22",
            "type": "div",
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
